providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java

```diff
@@ -63,15 +64,25 @@ public FlagdProvider(final FlagdOptions options) {
     }
 
     @Override
-    public void initialize(EvaluationContext evaluationContext) throws Exception {
+    public synchronized void initialize(EvaluationContext evaluationContext) throws Exception {
+        if (this.initialized) {
+            return;
+        }
+
         this.evaluationContext = evaluationContext;
         this.flagResolver.init();
+        this.initialized = true;
     }
 
     @Override
-    public void shutdown() {
+    public synchronized void shutdown() {
+        if (!initialized){
```

Comment:
```suggestion
        if (!initialized) {
```

I think this is your lint failure.

Reply:
yeah, fixed with the latest rebase :)